lib/Runtime/Debug/DiagObjectModel.cpp

```diff
@@ -3837,4 +3880,195 @@ namespace Js
+            break;
+        case TypeIds_SIMDBool8x16:
+            pResolvedObject->obj = JavascriptBoolean::ToVar(value.i8[i], scriptContext);
+            break;
+        case TypeIds_SIMDBool16x8:
+            pResolvedObject->obj = JavascriptBoolean::ToVar(value.i16[i], scriptContext);
+            break;
+        case TypeIds_SIMDUint32x4:
+            pResolvedObject->obj = JavascriptNumber::ToVarWithCheck(value.u32[i], scriptContext);
+            break;
+        case TypeIds_SIMDUint8x16:
+            pResolvedObject->obj = JavascriptNumber::ToVarWithCheck(value.u8[i], scriptContext);
+            break;
+        case TypeIds_SIMDUint16x8:
+            pResolvedObject->obj = JavascriptNumber::ToVarWithCheck(value.u16[i], scriptContext);
+            break;
+        default:
+            AssertMsg(false, "Unexpected SIMD typeId");
+            return FALSE;
+        }
+
+        pResolvedObject->propId = Constants::NoProperty;
+        pResolvedObject->scriptContext = scriptContext;
+        pResolvedObject->typeId = simdTypeId;
+        pResolvedObject->objectDisplay = pResolvedObject->CreateDisplay();
+        pResolvedObject->objectDisplay->SetDefaultTypeAttribute(DBGPROP_ATTRIB_VALUE_READONLY | DBGPROP_ATTRIB_VALUE_IS_FAKE);
+        pResolvedObject->address = nullptr;
+
+        return TRUE;
+    }
+
+    //--------------------------
+    // RecyclableSimdObjectDisplay
+
+    template <typename simdType, typename simdWalker>
+    LPCWSTR RecyclableSimdObjectDisplay<simdType, simdWalker>::Type()
+    {
+        TypeId simdTypeId = JavascriptOperators::GetTypeId(instance);
+
+        switch (simdTypeId)
+        {
+        case TypeIds_SIMDInt32x4:
+            return  L"SIMD.Int32x4";
+        case TypeIds_SIMDFloat32x4:
+            return  L"SIMD.Float32x4";
+        case TypeIds_SIMDInt8x16:
+            return  L"SIMD.Int8x16";
+        case TypeIds_SIMDInt16x8:
+            return  L"SIMD.Int16x8";
+        case TypeIds_SIMDBool32x4:
+            return  L"SIMD.Bool32x4";
+        case TypeIds_SIMDBool8x16:
+            return  L"SIMD.Bool8x16";
+        case TypeIds_SIMDBool16x8:
+            return  L"SIMD.Bool16x8";
+        case TypeIds_SIMDUint32x4:
+            return  L"SIMD.Uint32x4";
+        case TypeIds_SIMDUint8x16:
+            return  L"SIMD.Uint8x16";
+        case TypeIds_SIMDUint16x8:
+            return  L"SIMD.Uint16x8";
+        default:
+            AssertMsg(false, "Unexpected SIMD typeId");
+            return nullptr;
+        }
+    }
+
+    template <typename simdType, typename simdWalker>
+    LPCWSTR RecyclableSimdObjectDisplay<simdType, simdWalker>::Value(int radix)
+    {
+        StringBuilder<ArenaAllocator>* builder = GetStringBuilder();
+        builder->Reset();
+
+        simdType* simd = simdType::FromVar(instance);
+        SIMDValue value = simd->GetValue();
+
+        TypeId simdTypeId = JavascriptOperators::GetTypeId(instance);
+        wchar_t stringBuffer[SIMD_STRING_BUFFER_MAX_NAME];
```

Comment:
I am saying instead of stack allocated buffer use
WCHAR* stringBuffer = AnewArray(GetArenaFromContext(scriptContext), WCHAR, SIMD_STRING_BUFFER_MAX_NAME);